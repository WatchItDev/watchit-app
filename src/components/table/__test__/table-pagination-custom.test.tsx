import { render, screen, fireEvent } from '@testing-library/react';
import TablePaginationCustom from '../table-pagination-custom';

describe('[COMPONENTS]: TablePaginationCustom component testing', () => {
  it('matches snapshot', () => {
    const { container } = render(
      <TablePaginationCustom
        count={100}
        page={0}
        rowsPerPage={10}
        onPageChange={() => {}}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('renders with default rowsPerPageOptions', () => {
    render(
      <TablePaginationCustom
        count={100}
        page={0}
        rowsPerPage={10}
        onPageChange={() => {}}
      />
    );

    const selectElement = screen.getByRole('combobox', { name: /rows per page/i });
    expect(selectElement).toBeInTheDocument();
  });

  it('does not display dense switch when onChangeDense is not provided', () => {
    render(
      <TablePaginationCustom
        count={100}
        page={0}
        rowsPerPage={10}
        onPageChange={() => {}}
      />
    );

    expect(screen.queryByText('Dense')).not.toBeInTheDocument();
  });

  it('displays dense switch when onChangeDense is provided', () => {
    render(
      <TablePaginationCustom
        count={100}
        page={0}
        rowsPerPage={10}
        onPageChange={() => {}}
        dense={false}
        onChangeDense={() => {}}
      />
    );

    expect(screen.getByText('Dense')).toBeInTheDocument();
  });

  it('calls onChangeDense when dense switch is clicked', async () => {
    const handleChangeDense = vi.fn();

    render(
      <TablePaginationCustom
        count={100}
        page={0}
        rowsPerPage={10}
        onPageChange={() => {}}
        dense={false}
        onChangeDense={handleChangeDense}
      />
    );

    fireEvent.click(screen.getByRole('checkbox', { name: 'Dense' }));
    expect(handleChangeDense).toHaveBeenCalled();
  });

  it('applies custom sx prop to container', () => {
    const { container } = render(
      <TablePaginationCustom
        count={100}
        page={0}
        rowsPerPage={10}
        onPageChange={() => {}}
        sx={{ backgroundColor: 'rgb(240, 240, 240)' }}
      />
    );

    const boxContainer = container.firstChild;
    expect(boxContainer).toHaveStyle('background-color: rgb(240, 240, 240)');
  });

  it('renders with custom rowsPerPageOptions', () => {
    render(
      <TablePaginationCustom
        count={100}
        page={0}
        rowsPerPage={20}
        onPageChange={() => {}}
        rowsPerPageOptions={[20, 40, 60]}
      />
    );

    expect(screen.getByText('1â€“20 of 100')).toBeInTheDocument();
  });

  it('calls onPageChange when page is changed', async () => {
    const handlePageChange = vi.fn();
    render(
      <TablePaginationCustom
        count={100}
        page={0}
        rowsPerPage={10}
        onPageChange={handlePageChange}
      />
    );

    fireEvent.click(screen.getByRole('button', { name: /next page/i }));
    expect(handlePageChange).toHaveBeenCalled();
  });

  it('correctly displays when dense is true', () => {
    render(
      <TablePaginationCustom
        count={100}
        page={0}
        rowsPerPage={10}
        onPageChange={() => {}}
        dense={true}
        onChangeDense={() => {}}
      />
    );

    const denseSwitch = screen.getByRole('checkbox', { name: 'Dense' });
    expect(denseSwitch).toBeChecked();
  });
});
