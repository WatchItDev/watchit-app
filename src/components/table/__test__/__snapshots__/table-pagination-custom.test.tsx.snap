// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[COMPONENTS]: TablePaginationCustom component testing > matches snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-1kxonj9"
  >
    <div
      class="MuiTablePagination-root css-312mw9-MuiTablePagination-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-78c6dr-MuiToolbar-root-MuiTablePagination-toolbar"
      >
        <div
          class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
        />
        <p
          class="MuiTablePagination-selectLabel css-pdct74-MuiTablePagination-selectLabel"
          id=":r1:"
        >
          Rows per page:
        </p>
        <div
          class="MuiInputBase-root MuiInputBase-colorPrimary MuiTablePagination-input css-16c50h-MuiInputBase-root-MuiTablePagination-select"
        >
          <div
            aria-controls=":r2:"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby=":r1: :r0:"
            class="MuiSelect-select MuiTablePagination-select MuiSelect-standard MuiInputBase-input css-194a1fa-MuiSelect-select-MuiInputBase-input"
            id=":r0:"
            role="combobox"
            tabindex="0"
          >
            10
          </div>
          <input
            aria-hidden="true"
            aria-invalid="false"
            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            tabindex="-1"
            value="10"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiTablePagination-selectIcon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
        <p
          class="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
        >
          1â€“10 of 100
        </p>
        <div
          class="MuiTablePagination-actions"
        >
          <button
            aria-label="Go to previous page"
            class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
            disabled=""
            tabindex="-1"
            title="Go to previous page"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="KeyboardArrowLeftIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
              />
            </svg>
          </button>
          <button
            aria-label="Go to next page"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            title="Go to next page"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="KeyboardArrowRightIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
